ynr/apps/elections/uk/every_election.py

```diff
@@ -128,53 +128,86 @@ def get_or_create_partyset(self):
         return (self.party_set_object, self.party_set_created)
 
     def get_or_create_post(self):
+        """
+        Get or create a post based on the division and division set
+        from EveryElection
+
+        This is compex for a number of reasons:
+
+        1. Sometimes we have elections for organisations that don't have
+           sub-divisions. We need to make a post in that case anyway, so we use
+           information about the organisation to emulate a post here.
+
+        2. Some divisions have IDs that change over time, mainly because they
+           don't have GSS codes when we first see them. We need to detect this
+           case and update the IDs accordingly.
+        """
+
         if not self.parent or self.children:
             raise ValueError("Can't create YNR Post from a election group ID")
 
         # Make an organisation
-        self.get_or_create_organisation()
+        org, created = self.get_or_create_organisation()
 
         if self["division"]:
-            # Case 1, there is an organisational division relted to this
+            # Case 1, there is an organisational division related to this
             # post
-            slug = ":".join(
-                [
-                    self["division"]["division_type"],
-                    self["division"]["official_identifier"].split(":")[-1],
-                ]
-            )
+            slug = self["division"]["slug"]
             label = self["division"]["name"]
             role = self["division"]["official_identifier"]
```

Comment:
This line of code isn't new in this PR, but it also makes no sense (and the existing data in the `Post.role` column is unhelpful in most cases). This should probably really be `role = self["elected_role"]`. We can punt dealing with this to an issue if you want (my instinct is that given how little sense the data in `Post.role` makes, we can maybe bin the field rather than trying to fix it?), but I'm raising it here because I noticed it.